File: app/src/main/java/com/neome/java/api/ent/base/dto/DtoEntGroupActionPermissionMap.java
// neome.ai API - do not change
//

package com.neome.java.api.ent.base.dto;

import com.neome.java.api.meta.base.Types.MetaIdAction;

import org.jetbrains.annotations.Nullable;

import java.util.Map;

@SuppressWarnings("unused")
public class DtoEntGroupActionPermissionMap
{
  @Nullable
  public MetaIdAction defaultActionId;

  @Nullable
  public MetaIdAction[] defaultPinnedActionIdSet;

  @Nullable
  public Boolean hideActionMenu;

  public MetaIdAction[] keys;

  public Map<MetaIdAction, DtoEntActionPermission> map;

  @Nullable
  public MetaIdAction[] mobilePinnedActionIdSet;
}

File: app/src/main/java/com/neome/api/ent/base/dto/DtoEntGroupActionPermissionMap.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.ent.base.dto

import com.neome.api.meta.base.Types.MetaIdAction

interface DtoEntGroupActionPermissionMap {
    val defaultActionId: MetaIdAction?
    val defaultPinnedActionIdSet: List<MetaIdAction>?
    val hideActionMenu: Boolean?
    val keys: List<MetaIdAction>
    val map: Map<MetaIdAction, DtoEntActionPermission>
    val mobilePinnedActionIdSet: List<MetaIdAction>?
}
